<a href="https://colab.research.google.com/github/virajvaidya/ValueAtRiskModel/blob/main/AUSONG.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install yfinance

from scipy.stats import norm

import yfinance          as yf
import pandas            as pd
import numpy             as np
import datetime          as dt
import matplotlib.pyplot as plt

import requests


In [18]:
org = yf.Ticker("ORG.AX") #ORIGIN ENERGY
bhp = yf.Ticker("BHP.AX") #BHP
sto = yf.Ticker("STO.AX") #SANTOS
wpl = yf.Ticker("WPL.AX") #WOODSIDE PETROLEUM

#Stock info

In [ ]:
org.info

In [ ]:
bhp.info

In [ ]:
sto.info

In [ ]:
wpl.info

#Cashflow

In [ ]:
org.cashflow

In [ ]:
bhp.cashflow

In [ ]:
sto.cashflow

In [ ]:
wpl.cashflow

#Balance Sheet

In [ ]:
org.balance_sheet

In [ ]:
bhp.balance_sheet

In [ ]:
sto.balance_sheet

In [ ]:
wpl.balance_sheet

#Earnings

In [ ]:
org.earnings

In [ ]:
bhp.earnings

In [ ]:
sto.earnings

In [ ]:
wpl.earnings

#Institutional Holders

In [ ]:
org.institutional_holders

In [ ]:
bhp.institutional_holders

In [ ]:
sto.institutional_holders

In [ ]:
wpl.institutional_holders

#Datasets

In [ ]:
orgdata=yf.download("ORG.AX", period='max')
bhpdata=yf.download("BHP.AX", period='max')
stodata=yf.download("STO.AX", period='max')
wpldata=yf.download("WPL.AX", period='max')

In [ ]:
orgdata.head()

In [ ]:
bhpdata.head()

In [ ]:
stodata.head()

In [ ]:
wpldata.head()

#5-year stock price graphs

In [ ]:
orgticker = yf.Ticker('ORG.AX')
org_df    = orgticker.history(period="5y")
org_df.rename(columns = {'Close':'ORG Close'}, inplace = True)
org_df['ORG Close'].plot(title="Origin Energy Stock Price")

In [ ]:
bhpticker = yf.Ticker('BHP.AX')
bhp_df    = bhpticker.history(period="5y")
bhp_df.rename(columns = {'Close':'BHP Close'}, inplace = True)
bhp_df['BHP Close'].plot(title="BHP Stock Price")

In [ ]:
stoticker = yf.Ticker('STO.AX')
sto_df    = stoticker.history(period="5y")
sto_df.rename(columns = {'Close':'STO Close'}, inplace = True)
sto_df['STO Close'].plot(title="Santos Stock Price")

In [ ]:
wplticker = yf.Ticker('WPL.AX')
wpl_df    = wplticker.history(period="5y")
wpl_df.rename(columns = {'Close':'WPL Close'}, inplace = True)
wpl_df['WPL Close'].plot(title="Woodside Petroleum Stock Price")

In [54]:
import pandas as pd
df = pd.concat([org_df['ORG Close'], bhp_df['BHP Close'], sto_df['STO Close'],wpl_df['WPL Close']], axis=1)

In [ ]:
df.head()

In [ ]:
df.tail()

#Plotting the 5-year stock prices for comparison

In [ ]:
import matplotlib as plt
df.plot()
fig = plt.pyplot.gcf()
fig.set_size_inches(18.5, 10.5)

In [67]:
returns=df.pct_change()

In [ ]:
returns.tail()

#Setting up our fictitious portfolio of the 4 ONG companies

In [73]:
import numpy as np
weights   = np.array([.25, .25, .25, .25]) #Allocating equal weights to the stocks in our portfolio
initinv   = 1000000 #Initial investment in dollars

#Calculating the Covariance Matrix for VaR calculation

In [ ]:
cov_matrix = returns.cov()
cov_matrix

#Clacluating portfolio mean and standard deviation

In [83]:
avgreturns = returns.mean()
avgreturns

ORG Close    0.000158
BHP Close    0.000711
STO Close    0.000747
WPL Close    0.000078
dtype: float64

In [84]:
portfoliomean  = avgreturns.dot(weights)                           #Making the returns proportional with the weighting of our portfolio

In [85]:
portfoliostd = np.sqrt(weights.T.dot(cov_matrix).dot(weights))     #Computing the standard deviation of our portfolio after adjusting for weights

In [86]:
invmean        = (1+portfoliomean) * initinv                       #Computing the mean of our investment amount

In [87]:
invstd         = initinv * portfoliostd                            #Computing the standard deviation of our investment

#Computing the Value at Risk

In [88]:
from scipy.stats import norm

In [92]:
conf = 0.05                                                       #Asserting the confidence interval at 95%

In [93]:
confcutoff = norm.ppf(conf, invmean, invstd)                      #Defining the "cutoff" value where our 95% CI sits

In [94]:
var = initinv - confcutoff                                        #Calculating the VaR as Initial Investment value minus the cutoff

In [95]:
var

28195.09719076578

#Interpretation

The VaR value above of 28,195 means that:

We can say with 95% confidence that on an initial investment of AUD 1 million, our losses will NOT exceed AUD 28,195.